### DISCUSSION TITLE
Best way to approach DELLY BCFs

### DISCUSSION THREAD

**Discussion by lydiayliu** [uclahs-cds/pipeline-call-sSV#111] [Q&A] [answered]
I know the BCF format is in binary and meant to be storage-saving as compared to VCFs. However, I don't know how to preview a BCF file in the terminal and hence don't know how to use it for anything. My first step is always converting the BCF file into VCF using `bcftools`, which is not even a module available on the cluster.

I was wondering what the best way to approach BCF outputs of DELLY would be and what the design intention was in having it as the final output of the pipeline. If all users must convert it into VCF before analysis, it would seem natural for the pipeline to include such a step.

**Comment by tnyamaguchi** [comment, marked as answer]
@lydiayliu `bcftools/1.15.1` is available as an env module on the cluster. Please let me know if there is an issue with the module. Here is the usage:  `bcftools view [options] <in.vcf.gz>` and no need to convert the BCF to a new VCF file.

Regarding the pipeline design, it was inherited from the legacy Perl SV pipeline and we're considering using `vcf.gz` instead of `.bcf` as you suggested.

**Reply by lydiayliu** [reply]
Got it, I just realized that bcftools is available, thanks!

Thanks, I didn't see an issue for the consideration of adopting `vcf.gz` instead of `.bcf` so I posted a question here. I think it would be much more convienient. I still need to convert to a new VCF file because I need to load the VCFs into R for parsing. Unless there is a bcftools library for R that I'm not aware of.

**Reply by tnyamaguchi** [reply]
Yup, I agree. There is an issue already here - https://github.com/uclahs-cds/pipeline-call-sSV/issues/98. @Faizal-Eeman I believe this change should be relatively easy to add but can you comment on this?

**Reply by lydiayliu** [reply]
Ahh I thought the issue was considering converting `vcf.gz` to `bcf`, hence I was confused because I thought that would be creating additional hurdles for downstream users. Unless I was missing some simple way of working with BCF files, I didn't think that lesser file size would justify the trouble of conversion as per https://github.com/uclahs-cds/pipeline-call-sSV/discussions/97#discussioncomment-5245330.

Thanks for clarifying!

**Reply by Faizal-Eeman** [reply]
> Yup, I agree. There is an issue already here - https://github.com/uclahs-cds/pipeline-call-sSV/issues/98. @Faizal-Eeman I believe this change should be relatively easy to add but can you comment on this?

Yes, we can perhaps make this change in `6.0.0-rc.2`. I can prioritize it this week, so that the changes from current `rc.1` and `rc.2` could be added to meta-pipeline together.

**Reply by tnyamaguchi** [reply]
@lydiayliu Hmm, it seems like there was some miscommunication. I was thinking the other way around as you mentioned. #97

**Reply by tnyamaguchi** [reply]
@Faizal-Eeman @lydiayliu Also, if I recall, `call-gSV` has the same issue.

**Reply by lydiayliu** [reply]
@tyamaguchi-ucla Hmmm now I'm confused as well XD For the record, I vote for standardizing pipeline outputs in the `vcf.gz` format :) Thanks!

**Reply by Faizal-Eeman** [reply]
Discussion #97 and Issue #98 are old, and will be updated based on the best approach that's convenient for users.

So, we will go with a standardized `vcf.gz` output.
